AWSClientRuntime/Tests/Protocols/Ec2Query/Ec2ErrorRequestIdTests.swift

```diff
@@ -0,0 +1,73 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+        
+
+import ClientRuntime
+import SmithyTestUtil
+import XCTest
+@testable import AWSClientRuntime
+
+class Ec2ErrorRequestIdTests: XCTestCase {
```

Comment:
Can/should these tests be written as EC2-specific protocol tests?  If so what's the advantages or disadvantages of one approach over another?

Reply:
I don't see any reason to write these tests as protocol tests since the code being tested isn't generated.
I do think integration tests would be a nice addition here though, to test the same end-to-end flow I tested manually, but we currently don't have a mechanism for that. Once we do, this (like the other service specific bugs) are good candidates to add integration tests for. 

curious to hear your thoughts on this as well!